deployability/modules/allocation/vagrant/provider.py

```diff
@@ -171,10 +217,106 @@ def check_ip(ip):
             if response != 0:
                 return ip
 
-        for i in range(1, 255):
+        for i in range(2, 254):
             ip = f"192.168.57.{i}"
             if check_ip(ip):
                 return ip
 
         # If no available IP address is found, raise an exception.
         raise cls.ProvisioningError("No available IP address found.")
+
+    @staticmethod
+    def __macos_host(arch: str, action: str) -> str:
+        """
+        Returns the host parameters for macOS instances.
+
+        Args:
+            arch (str): The architecture of the instance.
+
+        Returns:
+            str: The provider name.
+        """
+        client = boto3.client('secretsmanager')
+        server_port = 22
+        timeout = 5
+        conn_ok = False
+        macos_host_parameters = {}
+
+        if arch == 'arm64':
+            try:
+                server_ip = client.get_secret_value(SecretId='devops_macstadium_m1_jenkins_ip')['SecretString']
+                ssh_password = client.get_secret_value(SecretId='devops_macstadium_m1_jenkins_password')['SecretString']
+                ssh_user = client.get_secret_value(SecretId='devops_macstadium_m1_jenkins_user')['SecretString']
```

Comment:
Why the Vagrant provider connects to AWS? (Asking just to know, as it is not clear for me)

Reply:
macOS deployments with Vagrant require a connection to Wazuh infrastructure (macStadium and Black Mini), the connection parameters (IP, user and password) were hosted in AWS Secret Manager, for this reason we use boto3 to connect and obtain this information